cmd/libsnap-confine-private/snap.c

```diff
@@ -30,7 +30,7 @@
 bool verify_security_tag(const char *security_tag, const char *snap_name)
 {
 	const char *whitelist_re =
-	    "^snap\\.([a-z](-?[a-z0-9])*)\\.([a-zA-Z0-9](-?[a-zA-Z0-9])*|hook\\.[a-z](-?[a-z])*)$";
+	    "^snap\\.([a-z0-9](-?[a-z0-9])*)\\.([a-zA-Z0-9](-?[a-zA-Z0-9])*|hook\\.[a-z](-?[a-z])*)$";
```

Comment:
i think this'll fail if a snap called `123test` (or `12to8`) ships a hook.

Reply:
d'oh, counted parens wrong.

Reply:
I added a test case to ensure it passes.